src/eterna/ui/Toolbar.ts

```diff
@@ -145,676 +386,3015 @@ export default class Toolbar extends ContainerObject {
+                        }
+                    });
+                } catch {
+                    this.topButtons.clear();
+                }
+                if (this.topButtons.size === 0) {
+                    this.initializeTopButtons();
+                }
+            }
+
+            this.addObject(
+                this.leftButtonsGroup,
+                this.topScrollContainer.content
+            );
+
+            const boosterBts: GameButton[] = [];
+            if (this._boostersData != null) {
+                this.boostersMenu = new ToolbarButton()
+                    .allStates(Bitmaps.smallImgBoosters)
+                    .setCategory(ButtonCategory.SOLVE)
+                    .setName('BoosterMenu')
+                    .disabled(undefined);
+                this._bottomButtons.set(
+                    'BoosterMenu',
+                    new ToolbarButton()
+                        .allStates(Bitmaps.smallImgBoosters)
+                    // .tooltip('Game options')
+                        .setCategory(ButtonCategory.SOLVE)
+                        .setName('BoosterMenu')
+                );
+
+                this._gameButtons.set('BoosterMenu', this.boostersMenu);
+                boosterBts.push(this.getMirrorButton(this.boostersMenu));
+                this.boostersMenu.display.interactive = true;
+                this.pushButtonToCategory(this.boostersMenu);
+
+                this.boostersMenu.clicked.connect(() => {
+                    if (this._boostersData != null && this._boostersData.actions != null) {
+                        const mode = this.mode as PoseEditMode;
+                        mode.showDialog(new BoosterDialog(this._boostersData));
+                    }
+                });
+
+                if (this._boostersData.paint_tools != null) {
+                    const mode: PoseEditMode = this.mode as PoseEditMode;
+                    for (const data of this._boostersData.paint_tools) {
+                        Booster.create(mode, data).then((booster) => {
+                            booster.onLoad();
+                            const button: GameButton = booster.createButton();
+                            button.setName(`BoosterPainting-${button.label}`)
+                                .setCategory(ButtonCategory.SOLVE);
+                            this.regs.add(
+                                button.clicked.connect(() => {
+                                    mode.setPosesColor(booster.toolColor);
+                                    this._deselectAllPaintTools();
+                                })
+                            );
+                            this.dynPaintTools.push(button);
+                            button.display.interactive = true;
+                            this.pushButtonToCategory(button);
+                        });
+                    }
+                }
+            }
+            this.boosterButtonsGroup = new ButtonsGroup(boosterBts);
```

Comment:
Why create a separate group for boosters? The booster button should just be like any other tool, able to be put into either the left or right bay

Reply:
fixed